tests/unit/test_charm.py

```diff
@@ -7,51 +7,61 @@
-            "services": {
-                "fastapi-service": {
-                    "override": "replace",
-                    "summary": "fastapi demo",
-                    "command": "uvicorn api_demo_server.app:app --host=0.0.0.0 --port=8000",
-                    "startup": "enabled",
-                    "environment": {
-                        "DEMO_SERVER_DB_HOST": None,
-                        "DEMO_SERVER_DB_PASSWORD": None,
-                        "DEMO_SERVER_DB_PORT": None,
-                        "DEMO_SERVER_DB_USER": None,
-                    },
-                }
+
+@pytest.fixture(autouse=True)
+def _patched_postgres_relation_data(monkeypatch):
+    def mock_return(*_):
+        return {}
+
+    monkeypatch.setattr(FastAPIDemoCharm, "fetch_postgres_relation_data", mock_return)
+
+
+@pytest.fixture(autouse=True)
+def _patched_version(monkeypatch):
+    monkeypatch.setattr(FastAPIDemoCharm, "version", "1.0.0")
+
+
+def test_pebble_layer(harness: ops.testing.Harness[FastAPIDemoCharm]):
+    # Expected plan after Pebble ready with default config
+    expected_plan = {
+        "services": {
+            "fastapi-service": {
+                "override": "replace",
+                "summary": "fastapi demo",
+                "command": "uvicorn api_demo_server.app:app --host=0.0.0.0 --port=8000",
+                "startup": "enabled",
+                "environment": {
+                    "DEMO_SERVER_DB_HOST": None,
+                    "DEMO_SERVER_DB_PASSWORD": None,
+                    "DEMO_SERVER_DB_USER": None,
+                    "DEMO_SERVER_DB_PORT": None,
+                },
             }
         }
-        mock_fetch_postgres_relation_data.return_value = {}
-        mock_version.return_value = "1.0.0"
-
-        # Simulate the container coming up and emission of pebble-ready event
-        self.harness.container_pebble_ready("demo-server")
-        # Get the plan now we've run PebbleReady
-        updated_plan = self.harness.get_container_pebble_plan("demo-server").to_dict()
-        # Check we've got the plan we expected
-        self.assertEqual(expected_plan, updated_plan)
-        # Check the service was started
-        service = self.harness.model.unit.get_container("demo-server").get_service(
-            "fastapi-service"
-        )
-        self.assertTrue(service.is_running())
-        # Ensure we set an ActiveStatus with no message
-        self.assertEqual(self.harness.model.unit.status, ops.ActiveStatus())
\ No newline at end of file
+    }
+
+    # Simulate the container coming up and emission of pebble-ready event
+    harness.container_pebble_ready("demo-server")
+    # Get the plan now we've run PebbleReady
+    updated_plan = harness.get_container_pebble_plan("demo-server").to_dict()
+    service = harness.model.unit.get_container("demo-server").get_service(
+        "fastapi-service"
+    )
+    status = harness.model.unit.status
+
+    # Check that we have the plan we expected
```

Comment:
I know this was there in the original, but taking the opportunity for a drive-by:

```suggestion
    # Check that we have the plan we expected:
```